File: insertion_sort.cpp
#include <iostream>
using namespace std;

void print(int arr[], int n) {
    for(int i=0;i<n;i++) cout << arr[i] << " ";
    cout << endl;
}

void insertionSort(int arr[], int n) {
    for(int i=1;i<n;i++) {
        cout << "Iterasi " << i << ": ";
        print(arr, n);

        int key = arr[i];
        int j = i - 1;

        while(j >= 0 && arr[j] > key) {
            arr[j+1] = arr[j];
            j--;
        }
        arr[j+1] = key;
    }
}

int main() {
    int n;
    cout << "Jumlah data: ";
    cin >> n;

    int arr[n];
    cout << "Masukkan data: ";
    for(int i=0;i<n;i++) cin >> arr[i];

    insertionSort(arr, n);

    cout << "Hasil sorting: ";
    print(arr, n);
}

File: python/insertion_sort.py
def insertion_sort(arr):
    for i in range(1, len(arr)):
        key = arr[i]
        j = i - 1

        print(f"Iterasi {i}: {arr}")

        while j >= 0 and key < arr[j]:
            arr[j+1] = arr[j]
            j -= 1
        arr[j+1] = key
    return arr

n = int(input("Jumlah data: "))
data = list(map(int, input().split()))

print("Hasil:", insertion_sort(data))
